packages/core/src/modules/member/use-cases/has-review-access.ts

```diff
@@ -0,0 +1,30 @@
+import { db } from '@oyster/db';
+
+export async function hasReviewAccess(id: string): Promise<boolean> {
+  try {
+    // Check if the student has any work experiences
+    const workExperienceOfStudent = await db
+      .selectFrom('workExperiences')
+      .select(['studentId'])
+      .where('studentId', '=', id)
+      .executeTakeFirst();
+
+    // If the student has no work experience, return true
+    if (!workExperienceOfStudent) {
+      return true;
+    }
+
+    // Check if the student has written any reviews
+    const companyReviewsFromStudent = await db
+      .selectFrom('companyReviews')
+      .select(['studentId'])
+      .where('studentId', '=', id)
+      .executeTakeFirst();
+
+    // Return true if a review is found, otherwise false
+    return !!companyReviewsFromStudent;
+  } catch (error) {
```

Comment:
Discussed with Rami-we do want to remove the try-catch block here. 

It is correct logic that we want to have error catching, but what happens is Oyster already uses [Sentry](https://sentry.io/welcome/) when errors arise. If we have a try-catch block, if an error arises it will be handled directly here rather than logged through Sentry.

Reply:
Will do, thanks!